src/System.ComponentModel.TypeConverter/src/System/ComponentModel/EnumConverter.cs

```diff
@@ -132,13 +139,20 @@ public override object ConvertTo(ITypeDescriptorContext context, CultureInfo cul
                     List<Enum> flagValues = new List<Enum>();
 
                     Array objValues = Enum.GetValues(EnumType);
+                    var isUnderlyingTypeUInt64 = Enum.GetUnderlyingType(EnumType) == typeof(ulong);
                     long[] ulValues = new long[objValues.Length];
                     for (int idx = 0; idx < objValues.Length; idx++)
                     {
-                        ulValues[idx] = Convert.ToInt64((Enum)objValues.GetValue(idx), culture);
+                        Enum enumVal = (Enum)objValues.GetValue(idx);
+                        ulValues[idx] = isUnderlyingTypeUInt64
+                            ? unchecked((long)Convert.ToUInt64(enumVal, culture))
+                            : Convert.ToInt64(enumVal, culture);
```

Comment:
Instead of repeating the ternary expression 4 times, consider a helper method like

```C#
private static long GetEnumValue(Enum enumVal, Culture culture, bool isUnderlyingTypeUInt64)
{
    return isUnderlyingTypeUInt64 ?
        unchecked((long)Convert.ToUInt64(enumVal, culture)) :
        Convert.ToInt64(enumVal, culture);
}
```

And then all of the "current" lines just change from Convert.ToInt64 to GetEnumValue.